addthis_block/src/Plugin/Block/AddThisBlock.php

```diff
@@ -26,173 +25,109 @@ class AddThisBlock extends BlockBase
-      '#title' => t('Buttons size'),
+
+    $form['settings']['addthis_settings']['type'] = array(
       '#type' => 'select',
-      '#default_value' => $this->configuration['buttons_size'],
-      '#options' => array(
-        'addthis_16x16_style' => t('Small (16x16)'),
-        'addthis_32x32_style' => t('Big (32x32)'),
+      '#title' => t('Formatter for @title', array('@title' => 'AddThis block')),
+      '#title_display' => 'invisible',
+      '#options' => $formatter_options,
+      '#default_value' => $settings['type'],
+      '#attributes' => array('class' => array('addthis-display-type')),
+      '#ajax' => array(
+        'callback' => array($this, 'addthisAjaxCallback'),
+        'wrapper' => 'addthis_type_settings',
       ),
     );
-    $form['settings']['addthis_settings']['counter_orientation'] = array(
-      '#title' => t('Counter orientation'),
-      '#description' => t('Specify the way service counters are oriented.'),
-      '#type' => 'select',
-      '#default_value' => $this->configuration['counter_orientation'],
-      '#options' => array(
-        'horizontal' => t('Horizontal'),
-        'vertical' => t('Vertical'),
-      )
-    );
-    $form['settings']['addthis_settings']['extra_css'] = array(
-      '#title' => t('Extra CSS declaration'),
-      '#type' => 'textfield',
-      '#size' => 40,
-      '#default_value' => $this->configuration['extra_css'],
-      '#description' => t('Specify extra CSS classes to apply to the toolbox'),
+    $form['settings']['addthis_settings']['type_settings'] = array(
+      '#prefix' => '<div id="addthis_type_settings"',
+      '#suffix' => '</div>',
     );
+    if($type == 'addthis_basic_toolbox'){
+      $basicToolbox = AddThis::getInstance()->getBasicToolboxForm($this, $settings['basic_toolbox']);
+      $form['settings']['addthis_settings']['type_settings']['basic_toolbox'] = $basicToolbox;
+    }
+    else if ($type == 'addthis_basic_button'){
+      $basicButton = AddThis::getInstance()->getBasicButtonForm($this, $settings['basic_button']);
+      $form['settings']['addthis_settings']['type_settings']['basic_button'] = $basicButton;
+    }
+
+
     return $form;
   }
-  public function blockSubmit($form, FormStateInterface $form_state) {
-    $this->configuration['share_services'] = $form_state->getValue(['settings', 'addthis_settings', 'share_services']);
-    $this->configuration['buttons_size'] = $form_state->getValue(['settings', 'addthis_settings', 'buttons_size']);
-    $this->configuration['counter_orientation'] = $form_state->getValue(['settings', 'addthis_settings', 'counter_orientation']);
-    $this->configuration['extra_css'] = $form_state->getValue(['settings', 'addthis_settings', 'extra_css']);
+
+  public function addthisAjaxCallback(array $form, FormStateInterface $form_state){
+    return $form['settings']['settings']['addthis_settings']['type_settings'];
+  }
+
+  public function blockSubmit($form, FormStateInterface $form_state)
```

Comment:
I'd start commenting these functions or use inheritdoc if it's a standard function in the class